core/app/[locale]/(default)/product/[slug]/_components/wishlist-button/index.tsx

```diff
@@ -0,0 +1,154 @@
+import { removeEdgesAndNodes } from '@bigcommerce/catalyst-client';
+import { getTranslations } from 'next-intl/server';
+import { cache } from 'react';
+
+import { Streamable } from '@/vibes/soul/lib/streamable';
+import { Favorite } from '@/vibes/soul/primitives/favorite';
+import { getSessionCustomerAccessToken, isLoggedIn } from '~/auth';
+import { client } from '~/client';
+import { graphql } from '~/client/graphql';
+import { revalidate } from '~/client/revalidate-target';
+import { TAGS } from '~/client/tags';
+import { ExistingResultType } from '~/client/util';
+
+import { WishlistButtonDropdown } from './dropdown';
+
+const wishlistButtonLimit = 50;
+const WishlistButtonQuery = graphql(`
+  query WishlistButtonQuery($first: Int, $productId: Int!) {
+    customer {
+      wishlistsContainingProduct: wishlists(
+        first: $first
+        filters: { productEntityIds: [$productId] }
+      ) {
+        edges {
+          node {
+            entityId
+            name
+            items(first: 50, filters: { productEntityIds: [$productId] }) {
+              edges {
+                node {
+                  entityId
+                  productEntityId
+                  variantEntityId
+                  product {
+                    sku
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+      wishlists(first: $first) {
+        edges {
+          node {
+            entityId
+            name
+          }
+        }
+      }
+    }
+  }
+`);
+
+const getWishlistButtonData = cache(async (productId: number, customerAccessToken?: string) => {
+  if (!customerAccessToken) {
+    return null;
+  }
+
+  const { data } = await client.fetch({
+    document: WishlistButtonQuery,
+    variables: { productId, first: wishlistButtonLimit },
+    customerAccessToken,
+    fetchOptions: { next: { revalidate, tags: [TAGS.customer] } },
```

Comment:
Makes sense to me that we cache this, but we usually don't cache customer queries. Maybe @bookernath has an opinion for this?

Reply:
Same as above, I guess so far our understand is that we don't want to cache customer related queries because it can store a lot of cache if you have many customers.

Reply:
@jorgemoya Yeah for sure. I'm fine either way! If we remove caching we can always re-add it later if it's causing performance issues.